src/functionsShellCommandAction.ts

```diff
@@ -49,6 +49,10 @@ export const actionFunction = async (options: Options) => {
   // If the port was not set by the --port flag or determined from 'firebase.json', just scan
   // up from 5000
   let port = 5000;
+  if (typeof options.port === "number") {
+    port = options.port;
```

Comment:
Note that in this implementation, if the specified flag is taken the CLI will gracefully use another port.

Still thinking whether that is the desired behavior or whether the `functions:shell` command should just error out if the specified port is available.

@aalej - what do think is the right behavior here?

Reply:
Hi @taeold, let me know if I may have misunderstood anything. The current implementation is that when a port is taken, the CLI will try to find an available open port.

If I run `firebase functions:shell --port 5005` and port 5005 is taken, the CLI will try to use port 5006 instead - if 5006 is open. I noticed that this behavior is also the same when the port specified in the firebase.json file is taken. E.g. if firebase.json contains the ff:
```
 "emulators": {
   "functions": {
     "port": 5001
   },
   "ui": {
     "enabled": true
   },
   "singleProjectMode": true
 }
```
Running `firebase functions:shell` when port 5001 is taken, the CLI will scan up from 5001 to find an open port.

I’m not sure what would be the best behavior here since currently, the PROS I see with keeping the current implementation is that we don’t have to manually find an open port to connect the emulator to, as the CLI automatically finds it for us. The CLI also provides a warning message similar to below if the specified port is taken(a similar message is also shown when running `firebase functions:shell --port <port_number>` when the specified port is taken):
```
⚠  functions: Functions Emulator unable to start on port 5001, starting on 5002 instead.
```

However, the CONS I see is that if we automatically find an open port, it might cause an issue if some service expects functions to run on the port the user specified. I do agree that in this scenario, it would be better to error out and show a message stating that the specified port is unavailable. 

It might be better if we could provide an option for the user to choose if they want to use an open port if the specified port is unavailable. E.g. if the specified port is taken, output a message similar to:
```
Functions Emulator unable to start on port <specified_port>, would you like to use <open_port> instead? (Y/n)
```

Reply:
Let's go ahead with this fix and see what our users think!